app/controllers/settings/ldap_test_controller.rb

```diff
@@ -0,0 +1,136 @@
+require "net/ldap"
+require "json"
+require "openssl"
+require "base64"
+
+# LdapTestController is used to validate LDAP Connector before saving
+class Settings::LdapTestController < ApplicationController
+  def ldap_filter_search(ldap, baseDN, filter_string)
+    filter = Net::LDAP::Filter.construct(filter_string)
+    attrs = ["mail", "cn", "sn", "objectclass"]
+    search_object = ldap.search(
+      base:          baseDN,
+      filter:        filter,
+      attributes:    attrs,
+      return_result: true
+    )
+    search_object
+  end
+
+  def render_result(result_hash, status, message)
+    result_hash[:result] = { status: status, message: message }
+    render json: result_hash
+  end
+
+  # Returns an ldap object
+  def get_ldap(host, port, start_tls, pem_string, anon_bind, dN, pass)
+    tls = if start_tls == "true"
+      :start_tls
+    else
+      :simple_tls
+    end
+
+    decoded_pem = Base64.decode64(pem_string)
+    cert = OpenSSL::X509::Certificate.new(decoded_pem)
+    cert_store = OpenSSL::X509::Store.new
+    cert_store.add_cert(cert)
+    tls_options = {
+      verify_mode: OpenSSL::SSL::VERIFY_PEER,
+      cert_store:  cert_store
+    }
+
+    auth_hash = if anon_bind == "true"
```

Comment:
Not a fan of this method of using if "a = b == c   if   x  else y  end" just looks very odd. Perhaps parentheses around the whole if block would help make it more clear what is happening? Would that work?